cwl/extract.cwl

```diff
@@ -77,4 +81,24 @@ steps:
            glob: "$(inputs.chroms).bam"
       baseCommand: [java.sh, org.stjude.compbio.sam.TweakSam]
 
+  # Step02c: extract unmapped read pairs
+  extract_unmapped:
+   in:
+     bam: bam
+   out: [unmapped_bam]
+   run: 
+     class: CommandLineTool
+     stdout: unmapped.bam
+     inputs: 
+       bam: 
+         type: File
+         inputBinding: 
+           position: 0
+     arguments: ["-h", "-F", "ref_name =~ /\\*/ and unmapped and mate_is_unmapped", "-f", "bam"]
```

Comment:
```suggestion
     arguments: ["-h", "-F", "ref_name =~ /\\*/", "-f", "bam"]
```
I think we want to pull out all no-ref reads.  There shouldn't be any no-ref reads that are mapped, so it shouldn't make a difference either way.  However, if there were mapped no-ref reads, then I'd rather pass them through unchanged.

Reply:
Does TweakSam's -x option require both reads to be mapped? My interpretation was the mate was either mapped to a different chromosome or unmapped.

Reply:
-x does require both reads to be mapped.  Read pairs with only one read mapped will be extracted via the chromosome-specific extractions, because unmapped mates (by convention--we should probably document this, as it's an assumption in our code!) retain the reference name of their mapped counterparts.  So, since the extraction by chromosome ignores mapping status and uses ref name only, it will pick those up.